CHARTER.md

```diff
@@ -0,0 +1,117 @@
+# Project Charter
+
+## Overview
+
+Crossplane is a framework for building cloud native control planes without
+needing to write code. It has a highly extensible backend that enables you to
+build a control plane that can orchestrate applications and infrastructure no
+matter where they run, and a highly configurable frontend that puts you in
+control of the schema of the declarative API it offers.
+
+## Control Planes
+
+Control planes first emerged in network packet switching, where devices must
+forward packets as fast as possible, but also be highly configurable. Breaking
+out the control plane from the data plane allows the former to be optimized for
+flexibility while the latter is optimized for speed. It also decouples the
+failure domains - a control plane failure does not prevent the data plane from
+functioning. Control planes are rooted in control theory, in particular closed
+loop control where the actual state of a system is observed in order to
+determine how best to drive it toward a desired state.
+
+Control planes power cloud computing. Cloud providers like AWS are built with
+control planes, as are projects like Kubernetes. In cloud computing the data
+plane is instead a “compute plane” consisting of applications and the
+infrastructure they run on - VMs, containers, databases, caches, queues, etc.
+
+Control planes are popular in cloud computing because they:
+
+* Are “always-on”, constantly correcting drift by driving observed state toward
+  desired state.
+* Are API-driven, which enables discovery, collaboration, and integration with
+  other systems.
+* Don’t need to be installed on every laptop and CI/CD system, making them
+  easier to operate.
+* Keep control in a separate failure domain from the plane under control (e.g.
+  compute).
+
+Cloud native organizations increasingly prefer control planes to approaches such
+as imperative scripts or Infrastructure-as-Code (IaC), but historically building
+a control plane has required a significant investment to design and code it from
+scratch. This is where Crossplane comes in.
+
+## What Crossplane Is
+
+Crossplane is a neutral place for vendors and individuals to come together in
+enabling control planes. It offers a framework to build control planes for cloud
+native computing without needing to write code (unless you want to). It shares a
+foundation with Kubernetes, but supports many powerful extensions that enable it
+to control anything - not just containers. The Crossplane project consists of:
+
+* The Crossplane Resource Model (XRM). An API model for Crossplane extensions.
+* The ability to compose bespoke control plane APIs for any compute plane.
+* A package format and manager to extend your control plane with:
+  * The ability to orchestrate any kind of compute plane provider.
+  * New ways to compose control plane APIs.
+  * New integration with supporting services such as secret stores.
```

Comment:
FWIW, here we are assuming secret stores plugins to be installed via package manager which is at least not the case with the [initial design](https://github.com/crossplane/crossplane/pull/3029).

Reply:
The idea is to give examples of non-provider extensions the package manager might include in future and Secret Stores was the "most real" example (webhooks and functions are others that aren't as far along). I'm not meaning to imply that this works today, but rather that doing something like this would be within the project's charter.